programs/cryptid/src/instructions/extend_transaction.rs

```diff
@@ -146,9 +143,14 @@ pub fn extend_transaction<'a, 'b, 'c, 'info>(
             unauthorized_signer,
             CryptidError::KeyMustBeSigner
         );
-    } else {
-        require!(!allow_unauthorized, CryptidError::KeyMustBeSigner)
     }
+
+    let allow_unauthorized_signer = ctx
```

Comment:
Yeah, those code is actually duplicated from other instructions. It just states that. IF there was a SPECIFIC unauthorized signer set in propose, we can ALLOW it in `get_cryptid_account_checked`, because it was already checked previously in the code.